cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -585,6 +586,102 @@ class SimpleArray
         }
     }
 
+    void sort(void)
+    {
+        if (ndim() != 1)
+        {
+            throw std::runtime_error("SimpleArray: Sorting is only supported in 1D array.");
+        }
+
+        std::sort(begin(), end());
+    }
+
+    SimpleArray<uint64_t> argsort(void)
+    {
+        if (ndim() != 1)
+        {
+            throw std::runtime_error("SimpleArray: Sorting is only supported in 1D array.");
+        }
+
+        SimpleArray<uint64_t> ret(shape());
+
+        { // Return array initialization
+            uint64_t cnt = 0;
+            std::for_each(ret.begin(), ret.end(), [&cnt](uint64_t & v)
+                          { v = cnt++; });
+        }
+
+        value_type const * buf = body();
+        auto cmp = [buf](uint64_t a, uint64_t b)
+        {
+            return buf[a] < buf[b];
+        };
+        std::sort(ret.begin(), ret.end(), cmp);
+        return ret;
+    }
+
+    void apply_argsort(SimpleArray<uint64_t> const & sorted_args)
+    {
+        if (ndim() != 1 || sorted_args.ndim() != 1)
+        {
+            throw std::runtime_error("SimpleArray: Sorting is only supported in 1D array.");
+        }
+        if (shape()[0] != sorted_args.shape()[0])
+        {
+            throw std::runtime_error("SimpleArray: argsort only support same shape");
+        }
+        if (shape()[0] < 2)
+        {
+            return;
+        }
+
+        std::vector<bool> applied_arg(shape()[0], false);
+
+        auto all = [](std::vector<bool> & vec)
+        {
+            for (auto i : vec)
+            {
+                if (i == false)
+                    return false;
+            }
+            return true;
+        };
+
+        auto next = [](std::vector<bool> & vec, ssize_t last)
+        {
+            for (ssize_t i = last; i < static_cast<ssize_t>(vec.size()); i++)
```

Comment:
Prefer pre-increment `++i` instead of post-increment `i++`.  They mean differently.